protocol/storage/client.go

```diff
@@ -56,6 +58,7 @@ type clientPorcelainAPI interface {
 	CreatePayments(ctx context.Context, config porcelain.CreatePaymentsParams) (*porcelain.CreatePaymentsReturn, error)
 	DealGet(context.Context, cid.Cid) (*storagedeal.Deal, error)
 	DAGGetFileSize(context.Context, cid.Cid) (uint64, error)
+	DAGCat(context.Context, cid.Cid) (io.Reader, error)
```

Comment:
Why does `DAGGetFileSize` belong in `clientPorcelainAPI` but `DAGCat` does not? I still don't have my "Should this go into porcelain?" heuristic dialed to the same setting as yours.

Reply:
Actually DAGGetFileSize should probably move into the porcelain call as well since it's used by the piece commitment logic. The point is that the process of putting together all the appropriate low level calls to determine the piece commitment of an imported piece is not something storage.Client should need to deal with. The whole point of porcelain is piecing together low-level calls to create a more humane interface.

I don't think Client should deal with DAG calls at all. I don't think the Reader -> tmp.File -> proofs.GeneratePieceCommitment belongs in proofs. This whole chain should be porcelain consumed by Client.

That would be better, but not better enough to be worth blocking this PR.